src/Data/Binary/Class.hs

```diff
@@ -239,6 +250,18 @@ instance Binary Integer where
                     let v = roll bytes
                     return $! if sign == (1 :: Word8) then v else - v
 
+-- | /Since: 0.7.x.y/
+#ifdef HAS_FIXED_CONSTRUCTOR
```

Comment:
No, in both cases `Integer` is serialised. I could add a test which is run when `MkFixed` is available, that they are actually the same.